plugin/trino-teradata/src/test/java/io/trino/plugin/teradata/ClearScapeManager.java

```diff
@@ -0,0 +1,202 @@
+            return new TeradataHttpClient(envUrl);
+        }
+        else {
+            throw new URISyntaxException(envUrl, "Provide valid environment URL");
+        }
+    }
+
+    /**
+     * Public method to create and start the ClearScape environment instance. Should be called
+     * before any operations requiring an active environment.
+     */
+    public void setup()
+    {
+        this.configJSON = new ObjectMapper().createObjectNode();
+        createAndStartClearScapeInstance();
+    }
+
+    /** Public method to stop the clearscape instance */
+    public void stop()
+    {
+        stopClearScapeInstance();
+    }
+
+    /**
+     * Public method to shut down and delete the ClearScape environment instance. Should be called
+     * to clean up resources after usage.
+     */
+    public void teardown()
+    {
+        shutdownAndDestroyClearScapeInstance();
+    }
+
+    /**
+     * Handles the logic for creating and starting a ClearScape environment instance. If the
+     * environment already exists and is stopped, it is started. If it doesn't exist, a new
+     * environment is created. Updates the {@code configJSON} with host/IP and authentication info.
+     */
+    private void createAndStartClearScapeInstance()
+    {
+        try {
+            TeradataHttpClient teradataHttpClient = getTeradataHttpClient(configJSON);
+            String name = ClearScapeEnvVariables.ENV_CLEARSCAPE_NAME;
+            String token = ClearScapeEnvVariables.ENV_CLEARSCAPE_TOKEN;
+
+            EnvironmentResponse response = null;
+            try {
+                response = teradataHttpClient.getEnvironment(new GetEnvironmentRequest(name), token);
+            }
+            catch (BaseException be) {
+                LOGGER.info("Environment {} is not available. {}", name, be.getMessage());
+            }
+
+            if (response == null || response.ip() == null) {
+                CreateEnvironmentRequest request = new CreateEnvironmentRequest(
+                        name,
+                        ClearScapeEnvVariables.ENV_CLEARSCAPE_REGION,
+                        ClearScapeEnvVariables.ENV_CLEARSCAPE_PASSWORD);
+                response = teradataHttpClient.createEnvironment(request, token).get();
+            }
+            else if (response.state() == EnvironmentResponse.State.STOPPED) {
+                EnvironmentRequest request = new EnvironmentRequest(name, new OperationRequest("start"));
+                teradataHttpClient.startEnvironment(request, token);
+            }
+
+            if (response != null) {
+                configJSON.put("host", response.ip());
+            }
+
+            ImmutableMap<String, Object> authMap = ImmutableMap.<String, Object>builder()
+                    .put(TeradataConstants.AUTH_TYPE, "TD2")
+                    .put("username", ClearScapeEnvVariables.ENV_CLEARSAOPE_USERNAME)
+                    .put("password", ClearScapeEnvVariables.ENV_CLEARSCAPE_PASSWORD)
```

Comment:
Generate random password using random string generation utility. don't commit sensitive information to git next time.